spec/p2p/reactor/p2p-api.md

```diff
@@ -0,0 +1,260 @@
+# API for Reactors
+
+The [`Switch`][switch-type] is the central component of the p2p layer implementation.
+It manages all the reactors running in a node, registered during the node setup phase
+(see the reactor's [Registration section][reactor-registration]),
+and keeps track of the connections with peers.
+
+## `Switch` API
+
+The table below summarizes the interaction of the standard reactors with the
+p2p layer, by listing the [`Switch`][switch-type] methods used by them:
+
+| `Switch` API method                                     | consensus | block sync | state sync | mempool | evidence  | PEX   |
+|--------------------------------------------|-----------|------------|------------|---------|-----------|-------|
+| `Peers() IPeerSet`                         | x         | x          |            |         |           | x     |
+| `NumPeers() (int, int, int)`               |           | x          |            |         |           | x     |
+| `Broadcast(Envelope) chan bool`            | x         | x          | x          |         |           |       |
+| `MarkPeerAsGood(Peer)`                     | x         |            |            |         |           |       |
+| `StopPeerForError(Peer, interface{})`      | x         | x          | x          | x       | x         | x     |
+| `StopPeerGracefully(Peer)`                 |           |            |            |         |           | x     |
```

Comment:
These functions (MarkPeerAsGood, StopPeer*) seem to imply non-trivial behavior (that also involve other reactors). This might be interesting to formalize (in quint).

Also, I understand that this is how it is right now. But would it be thinkable that these three functions move from the Switch API to the Peer API below? (I just try to understand the wiggle room)

Reply:
Connected to my above comment: I would like to understand whether in the future we would need the Switch API at all:
- maintaining the responsibility to maintain the information provided by Peers and NumPeers could be moved to the reactors
- broadcast could be implemented within the reactors using send
- Reactor may be replaced by a different mechanism to switch between protocols (syncing protocols to consensus).

Reply:
This discussion is being addressed here: https://github.com/cometbft/knowledge-base/pull/8

Reply:
The document referring to the separation of `Switch` and `Peer` is not a WIP, please refer to: https://github.com/cometbft/knowledge-base/blob/51eb728dae2f5535c902275297e4aa1e6a600531/p2p/reactors/switch-peer.md